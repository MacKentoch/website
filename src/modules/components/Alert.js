import styled from 'styled-components'
import { darken, lighten } from 'polished'
import theme from 'style/theme'

const Alert = styled.div`
  padding: 10px;
  border-radius: 4px;
  background-color: ${props => lighten(0.3, theme.colors[props.ui])};
  color: ${props => darken(0.2, theme.colors[props.ui])};
`

Alert.defaultProps = {
  ui: 'danger',
}

export default Alert
