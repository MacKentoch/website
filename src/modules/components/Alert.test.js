import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Alert from './Alert'

describe('Alert', () => {
  it('should render a div', () => {
    const wrapper = shallow(<Alert>Hello</Alert>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should render with ui: success', () => {
    const wrapper = shallow(<Alert ui="success">Hello</Alert>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
