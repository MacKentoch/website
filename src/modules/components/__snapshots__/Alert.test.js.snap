// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should render a div 1`] = `
<div
  className="sc-bdVaJa dUrZZD"
>
  Hello
</div>
`;

exports[`Alert should render with ui: success 1`] = `
<div
  className="sc-bdVaJa AFNnC"
>
  Hello
</div>
`;
